providers/keycloak.go

```diff
@@ -94,5 +94,17 @@ func (p *KeycloakProvider) EnrichSession(ctx context.Context, s *sessions.Sessio
 	}
 	s.Email = email
 
+	fullName, err := json.Get("name").String()
```

Comment:
What does Keycloak get now as the `session.User`? In other OIDC based providers, it is extracted from the ID Token subject claim (but this Keycloak provider is incorrectly using OAuth even though it should use OIDC).

Reply:
Unclear what the internal values were for `session.User`, as I wasn't getting this a value for this field back from the `/userinfo` endpoint at all. 

Could this have been affected by the mappings in Keycloak for my particular OIDC provider (CILogon)? I'm a bit fuzzy on how all of these scopes and mappings actually connect with the `/userinfo` produced, so I'm not sure how to answer your question.

Reply:
FYI - here's a draft PR we are working through to completely replace the existing Keycloak provider: https://github.com/oauth2-proxy/oauth2-proxy/pull/1107